### PR TITLE
(docs) DOCUMENT-1034 add detail exitcode

### PR DESCRIPTION
**Pull request by jbondpdx**
The usage notes for puppet-agent are accurate, but incomplete; but this PR adds more detail about the available exit codes.

### CODE REVIEW

File: lib/puppet/application/agent.rb [hunk 1]
```diff
@@ -124,18 +124,26 @@ def help
...
+and applies its configuration every 30 minutes.
 
-Some flags are meant specifically for interactive use -- in particular,
+Some flags are meant specifically for interactive use --- in particular,
 'test', 'tags' and 'fingerprint' are useful.
 
 '--test' does a single run in the foreground with verbose logging, then exits.
-It will also exit if it can't get a valid catalog. The exit code after running
-with '--test' is 0 if the catalog was successfully applied, and 1 if the run
+It alsos exit if it can't get a valid catalog. The exit code after running
```
Comment (natemccurdy) [inline lib/puppet/application/agent.rb]: alsos?

File: lib/puppet/application/agent.rb [hunk 2]
```diff
@@ -124,18 +124,26 @@ def help
...
+with '--test' is 0 if the catalog was successfully applied, or 1 if the run
```
Comment (natemccurdy) [inline lib/puppet/application/agent.rb]: The `--test` option includes the `--detailed-exitcodes` option. Maybe reword this sentence to reflect that?

The exit code when using `--test` is not a binary option of either `0` or `1`. It could be any of the exit codes specified by the `--detailed-exitcodes` option.

Comment (jbondpdx) [reply lib/puppet/application/agent.rb]: Hi @natemccurdy ; I can reword! So you're saying you don't have to pass --detailed-exitcodes at all with --test? It's just inherently included, yeah?

Oof, nice typo I added while I was at it, too. 🥇

Comment (natemccurdy) [reply lib/puppet/application/agent.rb]: Yes, that is correct `--test` bundles together multiple other flags. One of which is `--detailed-exitcodes`.

**Comment by jbondpdx** [comment]
Updated to fix typo, correctly apply present tense in some spots, and most importantly, give a correct clarification of possible exit codes.

File: lib/puppet/application/agent.rb [hunk 3]
```diff
@@ -124,31 +124,36 @@ def help
...
-with '--test' is 0 if the catalog was successfully applied, and 1 if the run
-either failed or wasn't attempted (due to another run already in progress).
+It also exits if it can't get a valid catalog. The exit code after running
+with '--test' is one of the following:
+
+* 0: The run succeeded with no changes or failures; the system was already in the desired state.
+* 1: The run failed, or wasn't attempted due to another run already in progress.
+* 2: The run succeeded, and some resources were changed.
+* 4: The run succeeded, and some resources failed.
+* 6: The run succeeded, and included both changes and failures.
```
Comment (pillarsdotnet) [inline lib/puppet/application/agent.rb]: I suggested replacing the above with:

```
After running with '--test', an exit code of 0 (success) means the system was
already in the desired state, because the catalog was applied with no changes
or failures. See the '--detailed-exitcodes' option for more information.
```

This is better, because:
* It explains the mnemonic of the `--test` attribute: what is being "tested" is whether the system is already in the desired state.
* It avoids repeating information that is available elsewhere on the same page.
* It clearly links `--test` with `--detailed-exitcodes` in the first and most prominent place that `--test` is mentioned.

Comment (pillarsdotnet) [reply lib/puppet/application/agent.rb]: @jbondpdx Ping?

Comment (jbondpdx) [reply lib/puppet/application/agent.rb]: Hi @pillarsdotnet , thank you for your feedback! Because --test inherently includes the --detailed-exitcodes option, I think it's worth calling out in this usage section, rather than making users look for that information further down the page. I have added an explicit mention of the --detailed-exitcodes option because I agree with you that they should have been more clearly linked.

**Comment by jbondpdx** [comment]
@puppetlabs/team-coremunity , let me know if you have other changes or would like me to squash commits before merge.

**Comment by joshcooper** [comment]
Commit 7171d608bbe5b8e1deb9016887e2d613563fb082 needs a `(docs)` prefix

**Comment by jbondpdx** [comment]
Thanks @joshcooper , sorry about forgetting the prefix!
